src/shared/schema/avl.schema.ts

```diff
@@ -170,6 +170,54 @@ const makeFilteredVehicleActivityArraySchema = (namespace: string, errors?: AvlV
         });
     }, z.array(vehicleActivitySchema));
 
+export const vehicleActivityCancellationSchema = z.object({
+    RecordedAtTime: z.string().min(1),
+    VehicleMonitoringRef: createNmTokenSiriValidation("VehicleMonitoringRef", false),
+    VehicleJourneyRef: z
+        .object({
+            DataFrameRef: createNmTokenOrNumberSiriValidation("DataFrameRef"),
+            DatedVehicleJourneyRef: createNmTokenOrNumberSiriValidation("DatedVehicleJourneyRef"),
+        })
+        .optional(),
+    LineRef: createNmTokenSiriValidation("LineRef", false),
+    DirectionRef: z.union([
+        z
+            .string()
+            .regex(NM_TOKEN_REGEX, {
+                message: `DirectionRef must be 1-${REQUEST_PARAM_MAX_LENGTH} characters and only contain letters, numbers, periods, hyphens, underscores and colons`,
+            })
+            .transform((direction) => directionMap[direction.toLowerCase()] ?? direction.toLowerCase()),
+        z.number(),
+    ]),
+});
+
+/**
+ * The purpose of this transformer is to filter out invalid AVL cancellation data
+ * and return the rest of the data as valid.
+ */
+const makeFilteredVehicleActivityCancellationArraySchema = (namespace: string) =>
+    z.preprocess((input) => {
+        const result = z.any().array().parse(input);
+
+        return result.filter((item) => {
+            const parsedItem = vehicleActivityCancellationSchema
+                .refine((activity) => getDate(activity.RecordedAtTime) <= getDate().add(1, "minute"), {
+                    message: "RecordedAtTime in future",
+                    path: ["RecordedAtTime"],
+                })
+                .safeParse(item);
+
+            if (!parsedItem.success) {
+                logger.warn("Error parsing item");
+                logger.warn(parsedItem.error.format());
+
+                putMetricData(`custom/${namespace}`, [{ MetricName: "MakeFilteredArraySchemaParseError", Value: 1 }]);
```

Comment:
I presume we'll want to capture a different metric for cancellation parse failures

```suggestion
                putMetricData(`custom/${namespace}`, [{ MetricName: "MakeFilteredCancellationArraySchemaParseError", Value: 1 }]);
```

Reply:
Instead of creating a different metric I created a different namespace called `SiriVmVehicleActivityCancellationSchema`